WordPress/Classes/ViewRelated/NUX/WPStyleGuide+Login.swift

```diff
@@ -0,0 +1,135 @@
+        onePasswordButton.sizeToFit()
+
+        textField.rightView = onePasswordButton
+        textField.rightViewPadding = UIOffset(horizontal: 20.0, vertical: 0.0)
+        textField.rightViewMode = .always
+
+        onePasswordButton.addTarget(target, action: selector, for: .touchUpInside)
+    }
+
+    /// Adds a 1password button to a stack view, if available
+    ///
+    class func configureOnePasswordButtonForStackView(_ stack: UIStackView, target: NSObject, selector: Selector) {
+        if !OnePasswordFacade().isOnePasswordEnabled() {
+            return
+        }
+
+        let onePasswordButton = UIButton(type: .custom)
+        onePasswordButton.setImage(UIImage(named: "onepassword-wp-button"), for: UIControlState())
+        onePasswordButton.sizeToFit()
+        onePasswordButton.setContentHuggingPriority(UILayoutPriorityRequired, for: .horizontal)
+        onePasswordButton.setContentCompressionResistancePriority(UILayoutPriorityRequired, for: .horizontal)
+
+        stack.addArrangedSubview(onePasswordButton)
+
+        onePasswordButton.addTarget(target, action: selector, for: .touchUpInside)
+    }
+
+    ///
+    ///
+    class func colorForErrorView(_ opaque: Bool) -> UIColor {
+        let alpha: CGFloat = opaque ? 1.0 : 0.95
+        return UIColor(fromRGBAColorWithRed: 17.0, green: 17.0, blue: 17.0, alpha: alpha)
+    }
+
+    ///
+    ///
+    class func edgeInsetForLoginTextFields() -> UIEdgeInsets {
+        return UIEdgeInsetsMake(7, 20, 7, 20)
+    }
+
+    /// Return the system font in medium weight for the given style
+    ///
+    /// - note: iOS won't return UIFontWeightMedium for dynamic system font :(
+    /// So instead get the dynamic font size, then ask for the non-dynamic font at that size
+    ///
+    class func mediumWeightFont(forStyle style: UIFontTextStyle) -> UIFont {
+        let fontToGetSize = WPStyleGuide.fontForTextStyle(style)
+        return UIFont.systemFont(ofSize: fontToGetSize.pointSize, weight: UIFontWeightMedium)
+    }
+
+    // MARK: - Google Signin Button Methods
+
+    /// A factory method for getting a button for Google Sign-in
+    ///
+    /// - Returns: A properly styled UIButton
+    ///
+    class func googleLoginButton() -> UIButton {
+        let baseString =  NSLocalizedString("Or you can {G} Log in with Google.", comment: "Label for button to log in using Google. The {G} will be replaced with the Google logo.")
+
+        let font = WPStyleGuide.mediumWeightFont(forStyle: .subheadline)
+        let button = UIButton()
+        button.clipsToBounds = true
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.contentHorizontalAlignment = .left
+        button.titleLabel?.font = font
+        button.titleLabel?.numberOfLines = 0
+        button.titleLabel?.lineBreakMode = .byWordWrapping
+
+        // Manually setting constraints to ensure that a multiline lable is fully
```

Comment:
I feel this comment is too long 😜 I know it's surprising that UIKit has this bug, but something like "These constraints are necessary to work around UIKit bugs with multiline buttons" does the job.